src/plugins/discover/public/application/main/hooks/use_url_tracking.ts

```diff
@@ -23,7 +24,12 @@ export function useUrlTracking(savedSearchContainer: DiscoverSavedSearchContaine
       if (!dataView) {
         return;
       }
-      const trackingEnabled = Boolean(dataView.isPersisted() || savedSearch.id);
+      const trackingEnabled =
+        // Disable for ad-hoc data views as it can't be restored after a page refresh
+        Boolean(dataView.isPersisted() || savedSearch.id) ||
+        // Enable for ES|QL, although it uses ad-hoc data views
+        isOfAggregateQueryType(savedSearch.searchSource.getField('query'));
+
       urlTracker.setTrackingEnabled(trackingEnabled);
```

Comment:
thx for adding these comments to clarify ... I think on the long run we should enable this functionality for ad-hoc data views (so we should store them e.g. in the browsers session behind the scene). Because the ad-hoc behavior was confusing me in my testing when going back to classic mode